apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/JVMUtil.java

```diff
@@ -0,0 +1,190 @@
+                classLoaders.add(clzz.getClassLoader());
+            }
+        }
+
+        for (ClassLoader classLoader : classLoaders) {
+            classResCalc(classLoader.getClass());
+        }
+    }
+
+    private static void classResCalc(Class c) {
+        try {
+            java.net.URL url = c.getResource("/");
+            if (null != url) {
+                JAR_PATH_MAP.put(url.toString(), "");
+            }
+        } catch (Exception e) {
+            LOGGER.warn("Error msg: {}", e.getMessage());
+        }
+    }
+
+    private static void loadJarFileList() {
+        String[] classPathJar = getPathSeparator().split(OSUtil.getPathSeparator());
+        for (String s : classPathJar) {
+            if (!s.endsWith(".jar")) {
+                continue;
+            }
+            String path = new File(s).getAbsolutePath();
+            JAR_FILE_LIST.add(path);
+        }
+        for (String s : JAR_PATH_MAP.keySet()) {
+            s = s.replace("file:", "");
+            if (s.endsWith("WEB-INF/classes/")) {
+                s = s.replace("WEB-INF/classes/", "WEB-INF/lib/");
+            }
+            File tempDir = new File(s);
+            if (!tempDir.exists() || !tempDir.isDirectory()) {
+                continue;
+            }
+            for (File file : tempDir.listFiles()) {
+                if (!file.getName().endsWith(".jar")) {
+                    continue;
+                }
+                JAR_FILE_LIST.add(file.getAbsolutePath());
+            }
+        }
+        loadSpringBootLib();
+    }
+
+    private static void loadSpringBootLib() {
+        List<String> springBootJars = new ArrayList<>();
+        for (String s : JAR_FILE_LIST) {
+            JarFile jar = null;
+            try {
+                jar = new JarFile(s);
+                if (jar.getManifest() == null || !"org.springframework.boot.loader.JarLauncher".equals(jar.getManifest().getMainAttributes().getValue("Main-Class"))) {
+                    continue;
+                }
+                Enumeration<JarEntry> jarEntry = jar.entries();
+                String jarPath;
+                while (jarEntry.hasMoreElements()) {
+                    jarPath = jarEntry.nextElement().getName();
+                    if (jarPath.endsWith(".jar")) {
+                        springBootJars.add(s + "!/" + jarPath);
+                    }
+                }
+
+            } catch (Exception e) {
+                LOGGER.warn("Error msg: {}", e.getMessage());
+            } finally {
+                try {
+                    if (jar != null) {
+                        jar.close();
+                    }
+                } catch (IOException e) {
+                    LOGGER.warn("Error msg: {}", e.getMessage());
+                }
+            }
+        }
+        if (springBootJars.size() > 0) {
+            JAR_FILE_LIST.addAll(springBootJars);
+        }
+    }
+
+    private static String getPathSeparator() {
```

Comment:
This method name doesn't match what it actually does

Reply:
I'll adjust the name.